### PR TITLE
Add markdown-table-formatter

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

* add `markdown-table-formatter` to CI https://github.com/nvuillam/markdown-table-formatter
* add `markdown-table-formatter-check` and `markdown-table-formatter-format` makefile targets

---

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Review by perk-sumo — APPROVED** [review]
🎸

**Comment by sumo-drosiek** [comment]
Can we stick to left side alignment?

File: deploy/README.md [hunk 1]
```diff
@@ -45,27 +45,28 @@ The diagram below illustrates the components of the Kubernetes collection soluti
...
-kubectl | 1.16.0
+| Name          |        Version         |
+|---------------|:----------------------:|
+| K8s with EKS  | 1.16<br/>1.17<br/>1.18 |
+| K8s with Kops | 1.16<br/>1.17<br/>1.18 |
+| K8s with GKE  |          1.16          |
+| K8s with AKS  | 1.16<br/>1.17<br/>1.18 |
+| OpenShift     |          4.6           |
+| Helm          |     3.4.0 (Linux)      |
+| kubectl       |         1.16.0         |
```
Comment (sumo-drosiek) [inline deploy/README.md:65]: It doesn't look like very readable with center alignment

File: deploy/README.md [hunk 2]
```diff
@@ -45,27 +45,28 @@ The diagram below illustrates the components of the Kubernetes collection soluti
...
 
 The following matrix displays the tested package versions for our Helm chart.
 
+<<<<<<< Updated upstream
```
Comment (pmalek-sumo) [inline deploy/README.md:69]: oops

File: deploy/README.md (hunk 1 above)
Comment (pmalek-sumo) [reply deploy/README.md:65]: I don't agree but I can revert this to please the crowd.

**Pull request merged**